subgraphs/enzyme-core/mappings/v4/ExternalPositionManager.ts

```diff
@@ -1809,6 +1824,163 @@ export function handleCallOnExternalPositionExecutedForFund(event: CallOnExterna
+        withdrawalToken,
+        withdrawalAmount,
+        denominationAsset,
+        'pendle-sell-pt',
+        event,
+      ).id;
+      change.market = pendleMarket.id;
+      change.save();
+    }
+
+    if (actionId == PendleV2ActionId.AddLiquidity) {
+      let decoded = ethereum.decode(
+        '(address,address,uint256,tuple(uint256,uint256,uint256,uint256,uint256),uint256)',
+        event.params.actionArgs,
+      );
+
+      if (decoded == null) {
+        return;
+      }
+
+      let tuple = decoded.toTuple();
+
+      let market = tuple[0].toAddress();
+      let depositToken = ensureAsset(tuple[1].toAddress());
+      let depositAmount = toBigDecimal(tuple[2].toBigInt(), depositToken.decimals);
+
+      let pendleMarket = ensurePendleV2AllowedMarket(vault, market);
+
+      let position = usePendleV2Position(event.address.toHex());
+      position.lpTokenHoldings = arrayUnique(position.lpTokenHoldings.concat([pendleMarket.id]));
+      position.save();
+
+      let change = createPendleV2PositionChange(event.params.externalPosition, 'AddLiquidity', vault, event);
+      change.assetAmount = createAssetAmount(
+        depositToken,
+        depositAmount,
+        denominationAsset,
+        'pendle-add-liquidity',
+        event,
+      ).id;
+      change.market = pendleMarket.id;
+      change.save();
+    }
+
+    if (actionId == PendleV2ActionId.RemoveLiquidity) {
+      let decoded = ethereum.decode('(address,address,uint256,uint256,uint256)', event.params.actionArgs);
+
+      if (decoded == null) {
+        return;
+      }
+
+      let tuple = decoded.toTuple();
+
+      let market = tuple[0].toAddress();
+      let withdrawalToken = ensureAsset(tuple[1].toAddress());
+      let withdrawalAmount = toBigDecimal(tuple[2].toBigInt(), withdrawalToken.decimals);
+
+      let pendleMarket = ensurePendleV2AllowedMarket(vault, market);
+
+      let balance = tokenBalance(market, event.params.externalPosition);
+      if (balance && balance.isZero()) {
+        let position = usePendleV2Position(event.params.externalPosition.toHex());
+        position.lpTokenHoldings = arrayDiff(position.lpTokenHoldings, [pendleMarket.id]);
+        position.save();
+      }
+
+      let change = createPendleV2PositionChange(event.params.externalPosition, 'RemoveLiquidity', vault, event);
+      change.assetAmount = createAssetAmount(
+        withdrawalToken,
+        withdrawalAmount,
+        denominationAsset,
+        'pendle-remove-liquidity',
+        event,
+      ).id;
+      change.market = pendleMarket.id;
+      change.save();
+    }
+
+    if (actionId == PendleV2ActionId.ClaimRewards) {
+      let decoded = ethereum.decode('(address[])', tuplePrefixBytes(event.params.actionArgs));
+
+      if (decoded == null) {
+        return;
+      }
+
+      let tuple = decoded.toTuple();
+
+      let assetAddresses = tuple[0].toAddressArray();
+
+      let assets = new Array<string>(assetAddresses.length);
```

Comment:
those are not assets but markets, we would have to iterate on markets and use `getRewardTokens()` to get all assets transferred to the vault. I'm thinking about standardizing the shape of position change across actions. So we could do:
`assets` - an array of all assets transferred to/from the vault 
`assetAmounts` - an array of all assets transferred to/from the vault (currently single element). That would be an empty array if amount cannot be accessed
`markets` - array of markets that actions interacted, so it would single element for all actions except the claimRewards